Can't run FreeRCT like described in the readme

I followed the instructions to download and compile FreeRCT, but trying to [run the game like described](https://github.com/FreeRCT/FreeRCT#running-the-program) doesn't work:

~~~~
$:> cd bin/
$:> ./freerct 
Failed creating directory '/home/kaputtnik/.local/share/freerct/save/' (File exists)
~~~~

The directory was created by the program, so i wonder why the message appears and exits. Next try:

~~~~
$:> ./freerct 
Data file data/font/Ubuntu-L.ttf is missing, the installation seems to be broken!
~~~~

Hm.... probably because i run from the `build/bin/` directory? The `data` folder is in the base directory (`FreeRCT/data/` after git clone) so i tried to run the binary from there:

~~~~
cd ../../
$:> build/bin/freerct 
Insufficient graphics loaded.
~~~~

Hm... maybe i have to use the `-u` switch:

~~~~
$:> cd build/bin/
$:> ./freerct -u /home/kaputtnik/Quellcode/FreeRCT/data/
Failed creating directory '/home/kaputtnik/Quellcode/FreeRCT/data//save/' (File exists)
~~~~

Ok, i know this message, next try:
~~~~
$:> ./freerct -u /home/kaputtnik/Quellcode/FreeRCT/
Data file data/font/Ubuntu-L.ttf is missing, the installation seems to be broken!
~~~~

And this also don't work for me:
~~~~
cd ../
$:> make run
Consolidate compiler generated dependencies of target rcdgen
[ 15%] Built target rcdgen
[ 23%] Built target rcd
Consolidate compiler generated dependencies of target freerct
[100%] Built target freerct
Data file data/font/Ubuntu-L.ttf is missing, the installation seems to be broken!
make[3]: *** [CMakeFiles/run.dir/build.make:70: CMakeFiles/run] Fehler 1
make[2]: *** [CMakeFiles/Makefile2:160: CMakeFiles/run.dir/all] Fehler 2
make[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/run.dir/rule] Fehler 2
make: *** [Makefile:189: run] Fehler 2
~~~~

What do i miss?

OS: archlinux

--- a/README.rst
+++ b/README.rst
@@ -8,7 +8,7 @@ Building the program
 
 Currently only Linux is officially supported, although Windows and MacOS *should* work. Let us know if it's broken.
 
-Almost everything is written in C++, which means you need *g++* or *clang++* to compile it. FreeRCT uses C++11 features, so g++ 4.8+ or clang 3.3+ is recommended.
+Almost everything is written in C++, which means you need *g++* or *clang++* to compile it. FreeRCT uses C++17 features, so g++ 7+ or clang 6+ is recommended.
 In addition, you need:
 
 * *lex/flex* - Scanner generator for generating RCD input files. (optional)
@@ -126,8 +126,7 @@ Now run the program
 
 .. code-block:: bash
 
-        $ cd bin
-        $ ./freerct
+        $ ./bin/freerct
 
 or
 
@@ -138,6 +137,10 @@ or
 which should open a window containing the main menu (see also the pictures in the blog).
 
 Pressing 'q' quits the program.
+
+Use ``--help`` or ``-h`` to view available command-line options.
+
+A detailed manual for the game is available `online <https://freerct.net/manual>`_.
 
 Building Troubleshoot
 ---------------------

--- a/src/fileio.cpp
+++ b/src/fileio.cpp
@@ -266,9 +266,8 @@ void MakeDirectory(std::string path)
 	if (path.empty() || PathIsDirectory(path.c_str())) return;
 
 	/* Strip trailing path separators. */
-	const size_t dir_sep_len = strlen(DIR_SEP);
-	while (path.size() >= dir_sep_len && path.compare(path.size() - dir_sep_len, dir_sep_len, DIR_SEP) == 0) {
-		for (size_t i = 0; i < dir_sep_len; ++i) {
+	while (StrEndsWith(path.c_str(), DIR_SEP, false)) {
+		for (size_t i = strlen(DIR_SEP); i > 0; --i) {
 			path.pop_back();
 		}
 	}
@@ -346,7 +345,7 @@ const std::string &GetUserHomeDirectory()
  */
 std::string FindDataFile(const std::string &name)
 {
-	for (std::string path : {std::string(".."), freerct_install_prefix()}) {
+	for (std::string path : {std::string("."), std::string(".."), std::string("..") + DIR_SEP + "..", freerct_install_prefix()}) {
 		path += DIR_SEP;
 		path += name;
 		if (PathIsFile(path.c_str())) return path;

--- a/src/fileio.cpp
+++ b/src/fileio.cpp
@@ -261,10 +261,19 @@ bool RcdFileReader::GetBlob(void *address, size_t length)
  * @param path Path of the directory.
  * @todo At the time of writing (2021-06-30) this is tested only on Linux. Before using it anywhere else, test this on all platforms (especially Windows).
  */
-void MakeDirectory(const std::string &path)
+void MakeDirectory(std::string path)
 {
-	if (PathIsDirectory(path.c_str())) return;
+	if (path.empty() || PathIsDirectory(path.c_str())) return;
 
+	/* Strip trailing path separators. */
+	const size_t dir_sep_len = strlen(DIR_SEP);
+	while (path.size() >= dir_sep_len && path.compare(path.size() - dir_sep_len, dir_sep_len, DIR_SEP) == 0) {
+		for (size_t i = 0; i < dir_sep_len; ++i) {
+			path.pop_back();
+		}
+	}
+
+	/* Recursively create parent directories. */
 	const size_t sep_pos = path.rfind(DIR_SEP);
 	if (sep_pos != std::string::npos) MakeDirectory(path.substr(0, sep_pos));
 

--- a/src/fileio.h
+++ b/src/fileio.h
@@ -78,7 +78,7 @@ bool PathIsDirectory(const char *path);
 
 DirectoryReader *MakeDirectoryReader();
 
-void MakeDirectory(const std::string &path);
+void MakeDirectory(std::string path);
 void CopyBinaryFile(const char *src, const char *dest);
 
 const std::string &GetUserHomeDirectory();

--- a/src/rcdfile.cpp
+++ b/src/rcdfile.cpp
@@ -12,6 +12,7 @@
 #include "fileio.h"
 #include "string_func.h"
 #include "rev.h"
+#include <memory>
 
 RcdFileCollection _rcd_collection; ///< Available RCD files.
 
@@ -70,27 +71,41 @@ void RcdFileCollection::AddFile(const RcdFileInfo &rcd)
 /** Scan directories, looking for RCD files to add. */
 void RcdFileCollection::ScanDirectories()
 {
-	DirectoryReader *reader = MakeDirectoryReader();
-
 	const std::string _rcd_paths[] = {
-		"rcd",
+		".",
 		freerct_install_prefix() + DIR_SEP + "rcd",
-		"",
 	};
+	for (const std::string &rcd_path : _rcd_paths) this->ScanDirectory(rcd_path.c_str(), 3);
+}
 
-	const std::string *rcd_path = _rcd_paths;
-	while (!rcd_path->empty()) {
-		reader->OpenPath(rcd_path->c_str());
-		for (;;) {
-			const char *fname = reader->NextFile();
-			if (fname == nullptr) break;
-			if (!StrEndsWith(fname, ".rcd", false)) continue;
-			this->ScanFileForMetaInfo(fname);
+/**
+ * Recursively scan a directory, looking for RCD files to add.
+ * @param dir Directory to scan.
+ * @param recursion_depth Remaining layers of recursion depth.
+ */
+void RcdFileCollection::ScanDirectory(const char *dir, int recursion_depth)
+{
+	std::unique_ptr<DirectoryReader> reader(MakeDirectoryReader());
+
+	reader->OpenPath(dir);
+	for (;;) {
+		const char *fname = reader->NextEntry();
+		if (fname == nullptr) break;
+
+		std::string filename(fname);
+		const size_t sep_pos = filename.rfind(DIR_SEP);
+		filename = filename.substr(sep_pos + strlen(DIR_SEP));
+		if (filename.empty() || filename == "." || filename == "..") continue;
+
+		if (reader->EntryIsDirectory()) {
+			if (recursion_depth > 0) this->ScanDirectory(fname, recursion_depth - 1);
+			continue;
 		}
-		reader->ClosePath();
-		rcd_path++;
+
+		if (!StrEndsWith(fname, ".rcd", false)) continue;
+		this->ScanFileForMetaInfo(fname);
 	}
-	delete reader;
+	reader->ClosePath();
 }
 
 /**

--- a/src/rcdfile.h
+++ b/src/rcdfile.h
@@ -30,6 +30,7 @@ public:
 	RcdFileCollection();
 
 	void ScanDirectories();
+	void ScanDirectory(const char *dir, int recursion_depth);
 	void AddFile(const RcdFileInfo &rcd);
 
 	std::map<std::string, RcdFileInfo> rcdfiles; ///< Found unique RCD files, mapping of uri to the Rcd file information.

